deploy.sh

```diff
@@ -0,0 +1,123 @@
+        echo "Unknown environment $environment, use one of:"
+        ls deploy/ | egrep -v '^(base)'
+        exit 1
+    fi
+}
+
+build_image() {
+    # build container image for dev or prod environments
+
+    local environment="${1?No environment passed}"
+    local do_build="${2?No build argument passed}"
+
+    if [[ "$do_build" == "yes" ]]; then
+        if [[ "$environment" == "devel" ]]
+        then
+            # creates the image on minikube's docker daemon
+            eval $(minikube docker-env)
+            docker build -t buildpack-admission:latest .
+        else
+            # Build the container image on the docker-builder host (currently tools-docker-imagebuilder-01.tools.eqiad1.wikimedia.cloud).
+            docker build . -f Dockerfile -t docker-registry.tools.wmflabs.org/buildpack-admission:latest
+            # Push the image to the internal repo
+            docker push docker-registry.tools.wmflabs.org/buildpack-admission:latest
+            echo "Successfully built container image for toolforge/toolsbeta environments with exit code 0. \
+            To deploy,log into k8s control node with repository checked out and run './deploy.sh (toolforge or toolsbeta)'"
+            exit 0
+        fi
+    fi
+}
+
+regenerate_certs() {
+    # creates a new certificate, a CSR to sign it, and a k8s secret with the signed cert and key
+
+    local refresh_certs="${1?No refresh_certs argument passed}"
+    if [[ "$refresh_certs" == "yes" ]]; then
+        ./utils/regenerate_certs.sh
+    fi
+}
+
+deploy_generic() {
+    # deploy buildpack-admission-controller image to either dev or prod environments
+
+    local environment="${1?No environment passed}"
+    local refresh_certs="${2?No refresh_certs argument passed}"
+
+    if [[ "$environment" == "devel" ]]
+    then
+        # generate the patch to override the ca bundle with the k8s secret we just created
+        ./utils/realize_patch.sh deploy/devel/webhook.patch.yaml.tpl
+        # Deploy the dev environment
+        kubectl apply -k deploy/devel
+    else
+        if [[ "$refresh_certs" == "yes" ]]
+        then
+            # do rolling restart to ensure the new certs created by regenerate_certs is being used
+            kubectl rollout restart -n buildpack-admission deployment/buildpack-admission
+        else
+            kubectl --as=admin --as-group=system:master apply -k deploy/toolforge
+        fi
+    fi
+}
+
+
+main () {
+    local do_build="no"
+    local refresh_certs="no"
+
+    while getopts "hvcbif" option; do
+        case "${option}" in
+        h)
+            help
+            exit 0
+            ;;
+        b) do_build="yes";;
+        c) refresh_certs="yes";;
+        v) set -x;;
+        *)
+            echo "Wrong option $option"
+            help
+            exit 1
+            ;;
+        esac
+    done
+    shift $((OPTIND-1))
+
+    local environment="${1?No environment passed}"
```

Comment:
Here we probably should do the same as:
https://gerrit.wikimedia.org/r/c/labs/tools/registry-admission-webhook/+/841539

As the standard is that it should be deployable without passing any options

Reply:
I assume (without verifying) that `/etc/wmcs-project` is a file that is available on all hosts. I think it's brilliant!

Reply:
yep, on all VMs that are puppetized in cloud VPS.

Note that we are falling back to prod in case there's any error trying to read that file.